src/plugins/legacy/polygonRoi/polygonRoiToolBox.cpp

```diff
@@ -163,122 +145,216 @@ medAbstractData *polygonRoiToolBox::processOutput()
     return nullptr;
 }
 
+void polygonRoiToolBox::loadContoursIfPresent(medAbstractImageView *v, unsigned int layer)
+{
+    medAbstractData *data = v->layerData(layer);
+    if (data->identifier().contains("medContours") && v==currentView)
+    {
+        viewEventFilter->loadContours(data);
+        addNewCurve->setChecked(true);
+        v->removeLayer(layer);
+    }
+}
+
 void polygonRoiToolBox::updateView()
 {
-    medAbstractView *view = this->getWorkspace()->tabbedViewContainers()->getFirstSelectedContainerView();
-    medAbstractImageView *v = qobject_cast<medAbstractImageView*>(view);
-    if (view)
+    medTabbedViewContainers *containers = this->getWorkspace()->tabbedViewContainers();
+    if (containers)
     {
-        // Toolbox does not work with meshes or vector images
+        if ( containers->currentIndex() == -1 )
+            return;
+        QList<medAbstractView*> viewsInTabSelected = containers->viewsInTab(containers->currentIndex());
+
+        medAbstractView *view = containers->getFirstSelectedContainerView();
+        if (!view)
+            return;
+        medAbstractImageView *v = qobject_cast<medAbstractImageView*>(view);
+        if (!v)
+            return;
+
+        QList<medViewContainer*> containersInTab = containers->containersInTab(containers->currentIndex());
+        if (viewsInTabSelected.size() > 1)
+        {
+            for (unsigned int i=0; i<v->layersCount(); ++i)
+            {
+                loadContoursIfPresent(v, i);
+            }
+            return;
+        }
+
+        if (v->layersCount()==1 && viewEventFilter)
+        {
+            medAbstractData *data = v->layerData(0);
+            if (data->identifier().contains("medContours"))
+                return;
+        }
+
         for (unsigned int i=0; i<v->layersCount(); ++i)
         {
             medAbstractData *data = v->layerData(i);
             if(!data || data->identifier().contains("vtkDataMesh")
-                    || data->identifier().contains("itkDataImageVector"))
+                                    || data->identifier().contains("itkDataImageVector"))
```

Comment:
?

Reply:
We can't process polygon ROI segmentation on a mesh. Only on image